ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -139,6 +154,29 @@ public void run() {
   }
 
   /**
+   * Pauses hard deletes. If hard deletes is half way in hard deleting a range, it will continue to finish hard deleting
+   * the range and then pause. Hard delete will be in paused state until {@link #resume()} is called.
+   */
+  void pause() throws InterruptedException {
+    reentrantLock.lock();
```

Comment:
This is to ensure that any on-going hard delete is complete. Acquiring the lock implicitly means that the hard deleter thread is either sleeping (after being caught up) or just about entering next cycle after completing one cycle.